.github/workflows/kflex-compatibility.yml

```diff
@@ -0,0 +1,100 @@
+name: KubeFlex CLI Compatibility Matrix
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  release:
+    types: [published]
+
+jobs:
+  build-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+
+      - name: Set up Python
+        uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065
+        with:
+          python-version: '3.11'
+
+      - name: Extract min kflex CLI version
+        id: minver
+        run: |
+          MIN_VERSION=$(python3 scripts/extract_min_kflex_version.py)
+          echo "min_kflex_version=$MIN_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Fetch all kflex CLI releases >= min version
+        id: set-matrix
+        run: |
+          MIN_VERSION=${{ steps.minver.outputs.min_kflex_version }}
+          # Get all release tags (vX.Y.Z)
+          VERSIONS=$(curl -s https://api.github.com/repos/kubestellar/kubeflex/releases | jq -r '.[].tag_name' | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V)
+          # Filter for >= min version
+          MATRIX_LIST=[]
+          for V in $VERSIONS; do
+            # Strip leading 'v' and compare
+            VNUM=${V#v}
+            # Compare using sort -V
+            if [ "$(printf "%s\n%s" "$MIN_VERSION" "$VNUM" | sort -V | head -1)" = "$MIN_VERSION" ]; then
```

Comment:
This is much more complicated than just testing `! [[ "$MIN_VERSION" < "$VNUM" ]]`